File: src/countDigits/countDigits.ex
defmodule Math do
   defp do_count_digits(0, count), do: count
   defp do_count_digits(number, count), do: number |> div(10) |> do_count_digits(count + 1)
   def count_digits(number) do
	number |> do_count_digits(0)
   end
end

File: src/countDigits/countdigits.c
/*C Program to count digits of a number using recursion*/
 
#include <stdio.h>
 
//function to count digits
int countDigits(int num)
{
    static int count=0;
     
    if(num>0)
    {
        count++;
        countDigits(num/10);
    }
    else
    {
        return count;
    }
}
int main()
{
    int number;
    int count=0;
     
    printf("Enter a positive integer number: ");
    scanf("%d",&number);
     
    count=countDigits(number);
     
    printf("Total digits in number %d is: %d\n",number,count);
     
    return 0;
}

